experimental/dds/tree2/src/core/tree/anchorSet.ts

```diff
@@ -470,79 +440,86 @@ export class AnchorSet implements ISubscribable<AnchorSetRootEvents> {
+		assert(nodes.length > 0, 0x409 /* Nodes must be non-empty */);
 
-			if (dst.parent !== undefined) {
-				dstPath = this.trackInner(dst.parent);
-			}
+		// There are anchors which are getting moved,
+		// therefor the destination needs to be created if it does not yet exist.
+		const dstPath = this.trackInner(dst.parent ?? this.root);
 
-			// Update moved items for new parent.
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			const offset = dst.parentIndex - srcStart!.parentIndex;
-			for (const moved of toMove) {
-				moved.parentIndex += offset;
-				moved.parentPath = dstPath ?? this.root;
-				moved.parentField = dst.parentField;
-			}
+		// Update nodes for new parent.
+		for (const node of nodes) {
+			node.parentIndex += offset;
+			node.parentPath = dstPath;
+			node.parentField = dst.parentField;
+		}
+
+		// Update new parent to add moved children
+		const field = dstPath.children.get(dst.parentField);
+		if (field === undefined) {
+			dstPath.children.set(dst.parentField, nodes);
 		} else {
-			// There are no anchors to move,
-			// therefor we want to avoid creating the destination if it does not already exist.
-			dstPath = this.find(dst.parent ?? this.root);
-			if (dstPath !== undefined) {
-				// Since we need a remove ref below to handle the `toMove.length > 0` case above,
-				// add a ref here so that does not break this case.
-				dstPath.addRef();
-			}
+			// Update existing field contents
+			const numberBeforeMove = this.increaseParentIndexes(field, dst.parentIndex, count);
+
+			// Insert toMove items into dstPath
+			// TODO: this will fail for very large numbers of anchors due to argument limits.
+			field.splice(numberBeforeMove, 0, ...nodes);
 		}
 
-		// Update dst
-		if (dstPath !== undefined) {
-			// Update new parent to add moved children
-			const field = dstPath.children.get(dst.parentField);
-			if (field === undefined) {
-				if (toMove.length > 0) {
-					dstPath.children.set(dst.parentField, toMove);
-				}
-			} else {
-				// Update existing field contents
-				let numberBeforeMove = 0;
-				let index = 0;
-				while (index < field.length && field[index].parentIndex < dst.parentIndex) {
-					numberBeforeMove++;
-					index++;
-				}
-				while (index < field.length) {
-					// Fix indexes in dst after moved items (add count).
-					field[index].parentIndex += count;
-					index++;
-				}
-				// Insert toMove items into dstPath
-				// TODO: this will fail for very large numbers of anchors due to argument limits.
-				field.splice(numberBeforeMove, 0, ...toMove);
-			}
+		dstPath.removeRef();
+	}
 
-			dstPath.removeRef();
+	/**
+	 * Updates the parent indexes of the given field to account for the given number of items being inserted at the given index.
+	 *
+	 * TODO: tests
+	 */
+	private increaseParentIndexes(
+		field: PathNode[],
+		fromParentIndex: number,
+		count: number,
+	): number {
```

Comment:
the return value from this function is not document. It should be documented (if needed) or removed if not.